src/store/index.js

```diff
@@ -85,5 +86,185 @@ export default createStore({
+          total_pages: 6,
+          current_page: 1,
+          has_next: false,
+          has_pre: false
+        },
+        currentPageProducts: []
+      },
+      mutations: {
+        setNewProductName (state, productName) {
+          state.newProduct.name = productName
+        },
+        setNewProductPrice (state, productPrice) {
+          state.newProduct.price = productPrice
+        },
+        setProductId (state) {
+          state.newProduct.id = uuidv4()
+        },
+        setAllProducts (state, originProducts) {
+          state.OriginProductsList = originProducts
+        },
+        setEditProduct (state, product) {
+          state.editingProduct = {
+            ...state.editingProduct,
+            ...product
+          }
+        },
+        clearProductInputs (state) {
+          state.newProduct.name = ''
+          state.newProduct.price = ''
+        },
+        clearProductList (state) {
+          state.newProductsList = []
+        },
+        updateProductsList (state, editedProduct) {
+          console.log(editedProduct)
+          state.newProductsList.forEach((item, index) => {
+            if (item.id === editedProduct.id) {
+              state.newProductsList.splice(index, 1, editedProduct)
+            }
+          })
+        },
+        updatOriginList (state, editedProduct) {
+          state.OriginProductsList.forEach((item, index) => {
+            if (item.id === editedProduct.id) {
+              state.OriginProductsList.splice(index, 1, editedProduct)
+            }
+          })
+          const allProductsListStr = JSON.stringify(state.OriginProductsList)
+          localStorage.setItem('allProducts', allProductsListStr)
+        },
+        deleteProduct (state, id) {
+          state.newProductsList.forEach((item, index) => {
+            if (item.id === id) {
+              state.newProductsList.splice(index, 1)
+            }
+          })
+        },
+        deleteOriginProduct (state, id) {
+          state.OriginProductsList.forEach((item, index) => {
+            if (item.id === id) {
+              state.OriginProductsList.splice(index, 1)
+            }
+          })
+          const allProductsListStr = JSON.stringify(state.OriginProductsList)
+          localStorage.setItem('allProducts', allProductsListStr)
+        },
+        setPagination (state, page) {
+          const limitLength = 10
+          const totalPages = Math.ceil(state.OriginProductsList.length / limitLength)
+          state.pagination.total_pages = totalPages
+          state.pagination.current_page = page
+          if (state.pagination.current_page === state.pagination.total_pages) {
+            state.pagination.has_next = false
+          } else {
+            state.pagination.has_next = true
+          }
+          if (state.pagination.current_page === 1) {
+            state.pagination.has_pre = false
+          } else {
+            state.pagination.has_pre = true
+          }
+        },
```

Comment:
可以再簡化寫法
```suggestion
        setPagination (state, page) {
          const limitLength = 10
          const totalPages = Math.ceil(state.OriginProductsList.length / limitLength)
          state.pagination.total_pages = totalPages
          state.pagination.current_page = page
          state.pagination.has_next = state.pagination.current_page >= state.pagination.total_pages
          state.pagination.has_pre = state.pagination.current_page !== 1
        },
```
還有幾個項目可注意:
* 像是 current_page 、total_pages , 通常我們會使用 [駝峰命名法](https://zh.wikipedia.org/wiki/%E9%A7%9D%E5%B3%B0%E5%BC%8F%E5%A4%A7%E5%B0%8F%E5%AF%AB) 
example: currentPage, totalPages